### PR TITLE
IPG-314 merge develop_cicd (new CI/CD) changes to relevant branches

### PR DESCRIPTION
**Pull request by zocimek**

### CODE REVIEW

**Review by ghost — DISMISSED** [review]
The bitbucket debug changes could be skipped...

File: lpwan-backend/pom.xml [hunk 1]
```diff
@@ -5,13 +5,11 @@
...
     <packaging>jar</packaging>
-    <version>1011.7.0-SNAPSHOT</version>
-    <scm>
-        <connection>scm:git:ssh://git@github.com/SoftwareAG/cumulocity-clients-java</connection>
-        <tag>HEAD</tag>
-    </scm>
+    <version>${revision}${changelist}</version>
 
     <properties>
+        <revision>1.206.0</revision>
```
Comment (ghost) [inline lpwan-backend/pom.xml:11]: Is this required?

File: README.md [hunk 2]
```diff
@@ -38,4 +38,8 @@ Please add [Cumulocity] [1] [Maven repository] [2] to Your `settings.xml` like t
   [1]: http://www.cumulocity.com
   [2]: http://maven.apache.org/
   [3]: https://www.cumulocity.com/guides
+  
+  ------------------------------
+
+These tools are provided as-is and without warranty or support. They do not constitute part of the Software AG product suite. Users are free to use, fork and modify them, subject to the license agreement. While Software AG welcomes contributions, we cannot guarantee to include every contribution in the master project.
```
Comment (ghost) [inline README.md]: This should be removed, cumulocity-clients-java is not a fan project

**Review by ghost — CHANGES_REQUESTED** [review]

File: .jenkins/scripts/test.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+set -e
+export MAVEN_OPTS="-Xmx4096m -XX:MetaspaceSize=1024m ${MAVEN_OPTS}"
+source ${BASH_SOURCE%/*}/common.sh
+
+call-mvn test -s $MVN_SETTINGS -Pci -U "$@"
+
+# MVN_EXEC="$(pwd)/mvnw"
```
Comment (d-mariak) [inline .jenkins/scripts/test.sh]: Do we need commented code?

File: .jenkins/bitbucket/sdk_test.jenkinsfile [hunk 4]
```diff
@@ -1,9 +1,23 @@
+import groovy.json.*
```
Comment (d-mariak) [inline .jenkins/bitbucket/sdk_test.jenkinsfile:1]: Same changes should be provided to sdk_test.jenkinsfile in github package?

Comment (ghost) [reply .jenkins/bitbucket/sdk_test.jenkinsfile:1]: Yeah I think that this is just not needed, as this is some test stuff. Probably shouldn't be in PR at all.

**Review by ghost — DISMISSED** [review]

File: README.md (hunk 2 above)
Comment (c8y-kh4mell) [inline README.md]: remove

File: microservice/pom.xml [hunk 5]
```diff
@@ -4,17 +4,16 @@
 
     <groupId>com.nsn.cumulocity.clients-java</groupId>
     <artifactId>microservice-dependencies</artifactId>
-    <version>1011.7.0-SNAPSHOT</version>
+    <version>${revision}${changelist}</version>
 
     <name>Cumulocity :: Clients Java :: Microservice</name>
     <description>Microservice</description>
     <packaging>pom</packaging>
-    <scm>
```
Comment (c8y-kh4mell) [inline microservice/pom.xml:12]: why removed?

File: .jenkins/scripts/common.sh [hunk 6]
```diff
@@ -11,29 +11,35 @@ add-key-for-host bitbucket.org
 add-key-for-host github.com
 add-key-for-host yum.cumulocity.com
 export resources=hudson@yum.cumulocity.com
-export release_args="-DskipTests -Dmaven.javadoc.skip=true -Dskip.microservice.package=false -Dskip.agent.package.container=false -Dnexus.url=http://nexus:8081  -Darguments=-Dskip.microservice.package=false -Dskip.agent.package.rpm=false -Dskip.agent.package.container=false -Dnexus.url=http://nexus:8081"
```
Comment (c8y-kh4mell) [inline .jenkins/scripts/common.sh:14]: why nexus.url removed?

**Review by c8y-kh4mell — DISMISSED** [review]

**Review by c8y-kh4mell — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**